CourseManager/Views/CourseDetailsView.swift

```diff
@@ -0,0 +1,147 @@
+//
+//  CourseDetailsView.swift
+//  CourseManager
+//
+//  Created by Summer Crow on 2022-10-03.
+//
+
+import SwiftUI
+
+struct CourseDetailsView: View {
+    var courseID: String
+    @StateObject var viewModel = CourseDetailsViewModel()
+    @State var showingAddCourseSheet: Bool = false
+    @State var isOpeningCourse = false
+    var view = ContentView()
```

Comment:
I think it is extra to store `view` here. May be it is associated with this bug
<img width="500" alt="Screenshot 2022-10-14 at 17 11 38" src="https://user-images.githubusercontent.com/23167917/195881527-08256a6a-074d-4f96-9a2a-73c09541aa75.png">

Reply:
I will be removing that var view, because I won't be needing it anymore.
As for the bug, I figured it out. I had a onAppear method on my View which calls another func to assign all the initial values to course. It should only be called once and then the details of the course edited by the user. But it was being called every time the statusPicker was closed. So it was overriding any changes to the course status. I just associated the view being opened with a Bool (editViewIsOpened) which is initially false and updated to true the first time the view appears only: 
  .onAppear(perform: {
                if !editViewIsOpened {
                setUpViewModelCourseInformationToEdit()
                }
            })